store/db_exchain.go

```diff
@@ -0,0 +1,298 @@
+package store
+
+import (
+	"io/ioutil"
+	"log"
+	"math"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/syndtr/goleveldb/leveldb/util"
+
+	dbm "github.com/tendermint/tm-db"
+)
+
+const Interval int64 = 100000
+
+type BlockDB struct {
+	db       dbm.DB
+	history  map[int64]dbm.DB
+	interval int64
+
+	name    string
+	backend dbm.BackendType
+	dir     string
+	hisDir  string
+
+	mtx sync.Mutex
+}
+
+var _ dbm.DB = (*BlockDB)(nil)
+
+func NewBlockDB(name string, backend dbm.BackendType, dir string) *BlockDB {
+	db := dbm.NewDB(name, backend, dir)
+
+	historyDBs := make(map[int64]dbm.DB)
+
+	hisDir := filepath.Join(dir, "block_history")
+	fs, err := ioutil.ReadDir(hisDir)
+	if err != nil && !os.IsNotExist(err) {
```

Comment:
how to process the case of folder block_history not exist

Reply:
!os.IsNotExist(err)